lib/pages/filter/model/filter.dart

```diff
@@ -60,6 +60,30 @@ class Filter {
     return list;
   }
 
+  List<FilterNode> findNodeByPath(List<String> path) {
+    List<FilterNode> result = [root];
+    if (path == null) {
+      return result;
+    }
+    path.forEach((element) {
+      FilterNode aux = _findNodeByPathHelper(element, result.last);
+      if (aux == null) {
+        return null;
+      }
+      result.add(aux);
+    });
+    return result;
+  }
+
+  FilterNode _findNodeByPathHelper(String nodeName, FilterNode parent) {
```

Comment:
Didn't I make a suggestion before about how this whole thing can be one line and doesn't need to be a function?